device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -298,6 +305,18 @@ public void open() throws TransportException
         log.debug("Amqp connection opened successfully");
     }
 
+    private void clearLocalState()
+    {
+        this.sessionHandlers.clear();
```

Comment:
Would there be any need to nullcheck anything in here? Just seeing it being called in the catch so it might give an unexpected NRE.

Reply:
Nope. Both of these queues are final, so they are instantiated once and never set to null